lib/bambora/json_request.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Bambora
+  class JSONRequest
+    attr_reader :client
+
+    def initialize(client)
+      @client = client
+      @path = ''
+    end
+
+    def request(method:, path:, params: {}, body: {})
+      parse_response(
```

Comment:
Do we need any error handling here? If there is a network Error, we'll get an informative error from Excon.

Based on the existing gem, is possible that we'll get an error response back from Bambora, but those errors are in JSON as well, so I think we should handle them in the application.

Reply:
i guess if server error happens(not application which will form correct json with error as the body) you won't get to JSON response instead you'll get JSON parse exception and this is our responsibility to handle it. Possible solution is to check for response code first and if it signals about error wrap it in valid JSON.